cargo-dist/src/tasks.rs

```diff
@@ -2484,6 +2488,115 @@ impl<'pkg_graph> DistGraphBuilder<'pkg_graph> {
         Ok(())
     }
 
+    fn add_pkg_installer(&mut self, to_release: ReleaseIdx) -> DistResult<()> {
+        if !self.local_artifacts_enabled() {
+            return Ok(());
+        }
+
+        // Clone info we need from the release to avoid borrowing across the loop
+        let release = self.release(to_release).clone();
+        let variants = release.variants.clone();
+        let bin_aliases = release.bin_aliases.clone();
+        let checksum = release.checksum;
+
+        // Make a pkg for every darwin platform
+        for variant_idx in variants {
+            let variant = self.variant(variant_idx);
+            let binaries = variant.binaries.clone();
+            let target = &variant.target;
+            if !target.contains("darwin") {
+                continue;
+            }
+
+            let variant_id = &variant.id;
+            let artifact_name = format!("{variant_id}.pkg");
+            let artifact_path = self.inner.dist_dir.join(&artifact_name);
+            let dir_name = format!("{variant_id}_pkg");
+            let dir_path = self.inner.dist_dir.join(&dir_name);
+
+            // Compute which package we're actually building, based on the binaries
+            let mut package_info: Option<(String, PackageIdx)> = None;
+            for &binary_idx in &binaries {
+                let binary = self.binary(binary_idx);
+                if let Some((existing_spec, _)) = &package_info {
+                    // we haven't set ourselves up to bundle multiple packages yet
+                    if existing_spec != &binary.pkg_spec {
+                        return Err(DistError::MultiPackage {
```

Comment:
...wild why was this logic exclusive to msi

Reply:
Oh right I think it's because I needed to enforce this for the sake of calling into cargo-wix, but I don't think the pkg stuff has that limitation?

Reply:
Oh, so the goal wasn't to create a single MSI containing every package but rather just that cargo-wix can't understand what to do when given multiple packages? I think I misunderstood the purpose of this guard then.

Reply:
Right, I see now why we've got a separate conflict here. We're fetching some package-level metadata to pass to the package build - specifically, the identifier field is package-level metadata. We don't have the package directly accessible to us - just on the binaries - and if we *did* end up with a conflict of which package they resolve to, I don't know if I could get a better answer than the MSI path does.